### ISSUE TITLE
config: refuses to --remove channel (URL) listed by --show

### ISSUE THREAD

**Issue by yarikoptic** [conda/conda#14360]
### Checklist

- [x] I added a descriptive title
- [x] I searched open reports and couldn't find a duplicate

### What happened?

I am trying to follow a workaround proposed in
- https://github.com/conda/conda-libmamba-solver/issues/283#issuecomment-1949560105
to overcome the problem with resolves whenever trying to install from conda-forge . More details for our particular use case with git-annex in
- https://github.com/datalad/datalad-installer/issues/206

But even though a channel (https://repo.anaconda.com/pkgs/main) is listed in config (by a URL):

```shell
❯ /tmp/mini1/bin/conda config --show channels
channels:
  - conda-forge
  - nodefaults
  - https://repo.anaconda.com/pkgs/main
  - https://repo.anaconda.com/pkgs/r
```
it cannot be removed:

```shell
❯ /tmp/mini1/bin/conda config --remove channels "https://repo.anaconda.com/pkgs/main"

CondaKeyError: 'channels': value 'https://repo.anaconda.com/pkgs/main' not present in config
```

which makes it for an odd UX.

### Conda list

❯ /tmp/mini1/bin/conda config --show-sources
==> /tmp/mini1/.condarc <==
channels:
  - https://repo.anaconda.com/pkgs/main
  - https://repo.anaconda.com/pkgs/r

==> /home/yoh/.condarc <==
channel_priority: strict
channels:
  - conda-forge
  - nodefaults

**Comment by travishathaway** [comment]
@yarikoptic,

I agree that this is confusing UX. As far as I can tell, `conda config` only tries to remove channel entries from `$HOME/.condarc`. These `.condarc` files can be located in many places though, including the root of the conda installation as shown by your example.

If I understand you correctly, you would like the expected behavior to be for conda to search through each `.condarc` file that shows up in `conda config --show-sources` and attempt to remove matching channel entries from each. I tend to agree with this but also realize this makes the operation of that command a little more complex, and it may produce more unexpected behavior that could confuse users.

I welcome a discussion here to determine if this really is the behavior we would like to see from `conda config --remove` going forward.

**Comment by pgunn** [comment]
It'd be interesting if there were some way to blacklist entries from other dotfiles, but maybe any site that's looking for a coordinated response to ContinuumIO's licensing madness would be better off actually finding and changing those shared dotfiles (and doing a DNS-based blocking, to be sure).

**Comment by yarikoptic** [comment]
Thank you @pgunn and @travishathaway !  As to me the issue lost its priority since I decided to make script proceed the way conda advised me in its error message -- by specifying alternative solver ;) FWIW, here was the "solution":

- https://github.com/datalad/datalad-installer/pull/210/files

so I am not too eager to explore the possibilities here, but would keep `--system` in mind in the future and hopefully that would come handy for some other strugglers.

**Comment by ykfq** [comment]
> [@yarikoptic](https://github.com/yarikoptic),
> 
> I agree that this is confusing UX. As far as I can tell, `conda config` only tries to remove channel entries from `$HOME/.condarc`. These `.condarc` files can be located in many places though, including the root of the conda installation as shown by your example.
> 
> If I understand you correctly, you would like the expected behavior to be for conda to search through each `.condarc` file that shows up in `conda config --show-sources` and attempt to remove matching channel entries from each. I tend to agree with this but also realize this makes the operation of that command a little more complex, and it may produce more unexpected behavior that could confuse users.
> 
> I welcome a discussion here to determine if this really is the behavior we would like to see from `conda config --remove` going forward.

Inspired by this comment, I find two `.condarc` and deleted the `~/miniconda3/.condarc` and I works!
